Src/ILGPU/Runtime/Cuda/CudaStream.cs

```diff
@@ -52,6 +53,21 @@ internal CudaStream(Accelerator accelerator)
                     StreamFlags.CU_STREAM_NON_BLOCKING));
         }
 
+        /// <summary>
+        /// Constructs a new cuda stream with given <see cref="StreamFlags"/>
+        /// </summary>
+        /// <param name="accelerator">The associated accelerator.</param>
+        /// <param name="flag">
+        /// Stream flag to use. Allows blocking and non-blocking streams.
+        /// </param>
+        internal CudaStream(Accelerator accelerator, StreamFlags flag)
```

Comment:
QUESTION: Should we add this method as-is? Or modify the existing `CreateStream(Accelerator)` method to accept `StreamFlags`?

Reply:
That is a good question. Maybe we want to make the blocking/non-blocking functionality generally available to all accelerators. The different platform-dependent implementations can then decide for themselves how to realize the intended behavior.

Reply:
I'm sure we can make this work on the CPU, but I'm skeptical on how we would get this to work on OpenCL. Isn't the whole idea of queues based on enqueued operations happening at some point and not when you enqueue them?

Reply:
I have no clear idea to the question, but to make things a bit clearer, when we create a stream without CU_STREAM_NON_BLOCKING, it does not create a "blocking" stream : the kernels executing in this stream are still asynchronous, BUT, from the CUDA documentation :
CU_STREAM_NON_BLOCKING: Specifies that work running in the created stream may run concurrently with work in stream 0 (the NULL stream), and that the created stream should perform no implicit synchronization with stream 0.
So if I understand correctly, the default behaviour of CUDA streams is a bit strange : they do implicit synchronization with stream 0 (the "default" stream), but not with other stream. I am not sure we want to emulate this strange behaviour to other accelerators. Maybe exposing public CudaStream CreateStream(StreamFlags flags) is enough.

Reply:
Looking at CL properties we can use, I don't see anything that would fit here to give the functionality as CUDA, maybe besides `CL_​QUEUE_​OUT_​OF_​ORDER_​EXEC_​MODE_​ENABLE`, but that's a stretch and would likely cause confusion.

Reply:
The `CudaStream` class currently has the constructor `internal CudaStream(Accelerator)`. My initial suggestion was, rather than adding a second constructor `internal CudaStream(Accelerator, StreamFlags)`, we just have a single constructor that accepts `StreamFlags`.

`StreamFlags` is currently only relevant to Cuda accelerators. Even the naming is Cuda-specific `CU_STREAM_DEFAULT` and `CU_STREAM_NON_BLOCKING`.

If we wanted to expose `StreamFlags`, I was thinking to add the method `public CudaStream CreateStream(StreamFlags)` to the `CudaAccelerator` class.

Reply:
Yup, that makes sense. Since the constructor is internal, this also wouldn't be a breaking API change.